pwndbg/aglib/disasm/arch.py

```diff
@@ -134,17 +135,39 @@ def memory_or_register_assign(left: str, right: str, mem_assign: bool) -> str:
     return memory_assign(left, right) if mem_assign else register_assign(left, right)
 
 
+# A registry of functions that create disassembly assistants for each architecture
+ALL_DISASSEMBLY_ASSISTANTS: Dict[str, Callable[[], DisassemblyAssistant]] = {}
+
+
+def register_disassembly_assistant(
+    arch: PWNDBG_SUPPORTED_ARCHITECTURES_TYPE, assistant: Callable[[], DisassemblyAssistant]
+):
+    ALL_DISASSEMBLY_ASSISTANTS[arch] = assistant
```

Comment:
If we keep this func, the `assistant` can be renamed to `create_assistant` or `new_assistant` or `get_assistant`?